identity-vault/startup-strategies/src/util/session-vault.ts

```diff
@@ -0,0 +1,110 @@
+import {
+  BrowserVault,
+  Vault,
+  VaultType,
+  DeviceSecurityType,
+  IdentityVaultConfig,
+} from '@ionic-enterprise/identity-vault';
+import { createVault } from './vault-factory';
+import { Session } from '../models/Session';
+
+export type UnlockMode = 'BiometricsWithPasscode' | 'InMemory' | 'SecureStorage';
+
+const vault: Vault | BrowserVault = createVault();
+let session: Session | null = null;
+let listeners: any[] = [];
+
+export const initializeVault = async (): Promise<void> => {
+  try {
+    await vault.initialize({
+      key: 'io.ionic.gettingstartedivreact',
+      type: VaultType.SecureStorage,
+      deviceSecurityType: DeviceSecurityType.None,
+      lockAfterBackgrounded: 2000,
+    });
+  } catch (e: unknown) {
+    await vault.clear();
+    await updateUnlockMode('SecureStorage');
+  }
+
+  vault.onLock(() => {
+    session = null;
+    emitChange();
+  });
+};
+
+export const getSnapshot = (): Session | null => {
+  return session;
+};
+
+export const subscribe = (listener: any) => {
```

Comment:
I think we should rename this and `getSnapshot()` to something that actually means something in the context of of but the `useSyncExternalStore()` hook _and_ the vault. Maybe:

- getSession()
- subscribeToSession()

Reply:
Not to complicate this, but in the Auth Playground demo I had settled on creating an isolated "store" to keep the state data within (see [here](https://github.com/ionic-enterprise/tutorials-and-demos-react/blob/main/demos/auth-playground/src/utils/authentication/store.ts)). This encapsulated the `useSyncExternalStore()` hook and related logic into a custom hook that could be used in components elsewhere in the app.

For code where you actually wanted to update the state data, the `setState()` method from that simple store would be used (like [here](https://github.com/ionic-enterprise/tutorials-and-demos-react/blob/main/demos/auth-playground/src/utils/authentication/authentication.ts#L56)).

If that approach is adopted here, then there'd be a custom hook now available, possibly named `useSession` that would always have the updated state data returned from it. You could use it in the Tabs component where you have the `useSyncExternalStore()` now.

Reply:
Can you check the mods I just made to `iv-getting-started` for this? If they look good, I will roll it into `iv-startup-strategies` as well, and then later update the actual tutorial as well.

Reply:
Those changes look good to me